src/libawkward/array/NumpyArray.cpp

```diff
@@ -4822,103 +4822,111 @@ namespace awkward {
+        ptr = as_type<uint8_t>(reinterpret_cast<uint8_t*>(contiguous_self.ptr().get()),
+                               contiguous_self.length(),
+                               dtype);
+        break;
+      case util::dtype::uint16:
+        ptr = as_type<uint16_t>(reinterpret_cast<uint16_t*>(contiguous_self.ptr().get()),
+                                contiguous_self.length(),
+                                dtype);
+        break;
+      case util::dtype::uint32:
+        ptr = as_type<uint32_t>(reinterpret_cast<uint32_t*>(contiguous_self.ptr().get()),
+                                contiguous_self.length(),
+                                dtype);
+        break;
+      case util::dtype::uint64:
+        ptr = as_type<uint64_t>(reinterpret_cast<uint64_t*>(contiguous_self.ptr().get()),
+                                contiguous_self.length(),
+                                dtype);
+        break;
+      case util::dtype::float16:
+        throw std::runtime_error("FIXME: numbers_to_type for float16 not implemented");
+      case util::dtype::float32:
+        ptr = as_type<float>(reinterpret_cast<float*>(contiguous_self.ptr().get()),
+                             contiguous_self.length(),
+                             dtype);
+        break;
+      case util::dtype::float64:
+        ptr = as_type<double>(reinterpret_cast<double*>(contiguous_self.ptr().get()),
                               contiguous_self.length(),
                               dtype);
-      break;
-    case util::dtype::float16:
-      throw std::runtime_error("FIXME: numbers_to_type for float16 not implemented");
-    case util::dtype::float32:
-      ptr = as_type<float>(reinterpret_cast<float*>(contiguous_self.ptr().get()),
-                           contiguous_self.length(),
-                           dtype);
-      break;
-    case util::dtype::float64:
-      ptr = as_type<double>(reinterpret_cast<double*>(contiguous_self.ptr().get()),
-                            contiguous_self.length(),
-                            dtype);
-      break;
-    case util::dtype::float128:
-      throw std::runtime_error("FIXME: numbers_to_type for float128 not implemented");
-    case util::dtype::complex64:
-      throw std::runtime_error("FIXME: numbers_to_type for complex64 not implemented");
-    case util::dtype::complex128:
-      throw std::runtime_error("FIXME: numbers_to_type for complex128 not implemented");
-    case util::dtype::complex256:
-      throw std::runtime_error("FIXME: numbers_to_type for complex256 not implemented");
-    default:
-      throw std::invalid_argument(
-        std::string("cannot recast NumpyArray with format \"")
-        + format_ + std::string("\""));
-    }
+        break;
+      case util::dtype::float128:
+        throw std::runtime_error("FIXME: numbers_to_type for float128 not implemented");
+      case util::dtype::complex64:
+        throw std::runtime_error("FIXME: numbers_to_type for complex64 not implemented");
```

Comment:
```suggestion
        throw std::runtime_error("FIXME: values_astype for complex64 not implemented");
```

(and similar for the others)